template-marketplace/hertzbeat-template-hub/src/main/java/org/apache/hertzbeat/templatehub/sureness/SurenessFilterExample.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.hertzbeat.templatehub.sureness;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.usthe.sureness.mgt.SurenessSecurityManager;
+import com.usthe.sureness.processor.exception.*;
+import com.usthe.sureness.subject.SubjectSum;
+import com.usthe.sureness.util.SurenessContextHolder;
+import jakarta.servlet.annotation.WebFilter;
+import org.apache.hertzbeat.templatehub.sureness.processor.RefreshExpiredTokenException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import jakarta.servlet.*;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * sureness filter class example, filter all http request
+ * @author tomsun28
+ * @date 23:22 2020-03-02
+ */
+@Order(1)
+@Component
+@WebFilter(filterName = "SurenessFilterExample", urlPatterns = "/*", asyncSupported = true)
+public class SurenessFilterExample implements Filter {
+
+    /** logger **/
+    private static final Logger logger = LoggerFactory.getLogger(SurenessFilterExample.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        logger.info("SurenessFilterExample init");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        logger.info("SurenessFilterExample doFilter");
+        try {
+            SubjectSum subject = SurenessSecurityManager.getInstance().checkIn(servletRequest);
```

Comment:
@tomsun28 The code encountered an exception here

Reply:
This problem is need use the jakarta_servlet creator instead of servlet creator. But suggest use `spring-boot3-starter-sureness` to ignore this config.